### ISSUE TITLE
Alarm changes made via panel not reporting

### ISSUE THREAD

**Issue by theyapps** [tsightler/ring-mqtt#654]
### Describe the Bug

Alarm changes made via panel not reporting

### Steps to Reproduce

1. Arm Alarm via Ring Panel (physical buttons) 
2. Entity "alarm_control_panel.josephine_alarm" still says disarmed
3. Disarm alarm, arm via ring app.
4. Entity "alarm_control_panel.josephine_alarm" still says armed
5. Disarm alarm, arm via Home assistant
6. Ring app and physical panel show the alarm as armed.
 
### Expected Behavior

alarm_control_panel.josephine_alarm should say armed

### Log Output

```shell
NA
```

### Config File

```shell
mqtt_url: mqtt://auto_username:auto_password@auto_hostname
mqtt_options: ""
livestream_user: ""
livestream_pass: ""
disarm_code: ""
enable_cameras: false
enable_modes: false
enable_panic: false
branch: addon
debug: ring-*
location_ids:
  - ""
snapshot_mode: disabled
beam_duration: 0
mqtt_host: <auto_detect>
mqtt_port: <auto_detect>
mqtt_user: <auto_detect>
mqtt_password: <auto_detect>
ring_token: ""
```

### Install Type

HAOS

### Version

5.4.0 

### Operating System

 Home Assistant 2023.6.3 Supervisor 2023.06.2 Operating System 10.3 Frontend 20230608.0 - latest 

### Architecture

 Arm64

### Machine Details

Pi 4

**Comment by michaelsleen** [comment]
I am experiencing the same issue.

**Comment by tsightler** [comment]
Yep, I didn't anticipate that the impulse types would be different between arming via the app vs the keypad, I assumed switch-mode impulse data would be the same in all cases.  Dev branch has an attempt to fix this is anyone is up for testing.

**Comment by theyapps** [comment]
> Yep, I didn't anticipate that the impulse types would be different between arming via the app vs the keypad, I assumed switch-mode impulse data would be the same in all cases. Dev branch has an attempt to fix this is anyone is up for testing.

I will try in the morning. Thanks.

**Comment by theyapps** [comment]
Is there a built in way of installing development versions of addons, or is it a matter of manual installation?

**Comment by tsightler** [comment]
Branch feature is what you are looking for:

https://github.com/tsightler/ring-mqtt/wiki/Installation-(Home-Assistant-Addon)#branch-selection

Just set branch config option to "dev" and restart, the addon will pull down the latest dev branch during startup and you can see it in the logs.  If you are using docker vs addon then it's "BRANCH" environment variable when starting the container, but same basic idea.

**Comment by theyapps** [comment]
Awesome, I have done that but then of course realized testing this when remote will not work due to reliance on physical panel so I have it installed now and will test it with the panel when I get home this afternoon.

**Comment by tsightler** [comment]
I've tried to test all the modes myself and I think I've got this working now, but having some verification would be useful before I publish the update.  Thanks for the report and testing!

**Comment by pimpinpanther96** [comment]
> I've tried to test all the modes myself and I think I've got this working now, but having some verification would be useful before I publish the update. Thanks for the report and testing!

Can confirm alarm mode is now updating in HA when changed from the keypad after setting branch to "dev"

**Comment by theyapps** [comment]
Agreed worked for me as well on dev branch

**Comment by tsightler** [comment]
Have released v5.4.1 which includes the fix for this.  Hopefully I've caught all the cases.  Closing this for now.
